### ISSUE TITLE
Underscore in domain name breaks whitelist editor

### ISSUE THREAD

**Issue by jspenguin2017** [gorhill/uBlock#3717]
I'm opening the issue here as I am confident that this is a valid issue of uBO. 

Step to reproduce: 
1. Go to `http://test_test-.test.jspenguin.com/` (only works on Windows) or `http://test_test.test.jspenguin.com/` (works on Ubuntu as well)
2. Whitelist the page with the popup pane
3. Go to whitelist tab in dashboard and see the line in red

![image](https://user-images.githubusercontent.com/7283682/38966385-31911a84-433f-11e8-98df-9dbcc7cfadcf.png)

This (only) prevents further editing of the whitelist. The whitelist itself is working as intended. I think it is a problem with these following lines in `ublock.js`: 
```js
µBlock.reWhitelistBadHostname = /[^a-z0-9.\-\[\]:]/;
µBlock.reWhitelistHostnameExtractor = /([a-z0-9\[][a-z0-9.\-]*[a-z0-9\]])(?::[\d*]+)?\/(?:[^\x00-\x20\/]|$)[^\x00-\x20]*$/;
```

After some testing, it appears that underscore (`_`) cannot be inside the host name (so `ex_ample.com` is not valid), but can be inside subdomains (so `_-.example.com` is valid, on Windows at least, `test_test.example.com` would be valid on Ubuntu)

I have updated my server to serve the test page for any subdomain of `test.jspenguin.com`, so you can test them out to see what is the proper RegExp. Note that `test.jspenguin.com` itself will redirect to the home page (due to laziness). 

@gorhill

**Comment by jspenguin2017** [comment]
Also I'm (currently) on Win 10 Chrome 68 uBlock Origin 1.15.24.

**Comment by gorhill** [comment]
> Maybe just lift the restriction and accept anything that looks remotely reasonable as valid?

Yes, as long as _a_ hostname can be extracted -- valid or not.

**Comment by gorhill** [comment]
Well since you are almost there, can you submit a pull request with the modified regexes?

**Comment by gorhill** [comment]
Fixed with PR #3717.
